src/pluto/image.cr

```diff
@@ -36,6 +36,33 @@ class Pluto::Image
     )
   end
 
+  private def each_channel(&)
+    yield @red, Channel::Red
+    yield @green, Channel::Green
+    yield @blue, Channel::Blue
+    yield @alpha, Channel::Alpha
+  end
+
+  private def [](ch : Channel)
+    case ch
+    when Channel::Red   then @red
+    when Channel::Green then @green
+    when Channel::Blue  then @blue
+    when Channel::Alpha then @alpha
+    else                     raise "Unknown channel #{ch} for Image"
+    end
+  end
+
+  private def []=(ch : Channel, channel : Array(UInt8))
+    case ch
+    when Channel::Red   then @red = channel
+    when Channel::Green then @green = channel
+    when Channel::Blue  then @blue = channel
+    when Channel::Alpha then @alpha = channel
+    else                     raise "Unknown channel #{ch} for Image"
+    end
+  end
```

Comment:
I made all of these methods private, but thinking about it, I can't see why they couldn't be public too. Not really sure who I was protecting when I did that.

Reply:
I think having them public is useful.

Reply:
Also maybe we need a better name for `Channel`? Not really sure what would be more fitting, but having both `ch` and `channel` feels a bit odd.

Reply:
Ha! Yeah, I struggled with that name too, but couldn't think of anything better. `ChannelType` might be more descriptive as an enum name, and then `ch` could rename to `type`?

Reply:
`ChannelType` is good. I would go with `channel_type` instead of just `type`.